### ISSUE TITLE
Docker Environment Variables / Arguments & Llama3.1 Support

### ISSUE THREAD

**Issue by plitc** [h2oai/h2ogpt#1767]
Hello H2OGPT Team,

can you please list a Docker example that exactly reflects the functionality of https://gpt.h2o.ai/?

We currently cannot load the new Llama3.1 and with the option: prompt_type=llama the model only spits out confusing stuff.

`root@ai-ubuntu22gpu-gpt:/opt# cat run-latest4`
`#!/bin/sh`
`export HUGGING_FACE_HUB_TOKEN=...PRIVATE...`
`export CUDA_VISIBLE_DEVICES="0,1,2,3"`
`docker run \`
`       --gpus='"device=0,1,2,3"' \`
`       --runtime=nvidia \`
`       --shm-size=16g \`
`       -p 7860:7860 \`
`       --rm --init \`
`       -v /etc/passwd:/etc/passwd:ro \`
`       -v /etc/group:/etc/group:ro \`
`       -u 1000:1000 \`
`       -v /opt/h2o_gpt_data/.cache:/workspace/.cache \`
`       -v /opt/h2o_gpt_data/save:/workspace/save \`
`       -v /opt/h2o_gpt_data/db_dir_UserData:/workspace/db_dir_UserData \`
`       -v /opt/h2o_gpt_data/tmp:/tmp \`
`       -v /opt/CEPHFS_DATA:/workspace/user_path/CACHE \`
`       -e HUGGING_FACE_HUB_TOKEN=$HUGGING_FACE_HUB_TOKEN \`
`       -e TOKENIZERS_PARALLELISM=false \`
`       -e ADMIN_PASS=...PRIVATE... \`
`       -e CONCURRENCY_COUNT=1 \`
`       -e API_OPEN=1 \`
`       -e ALLOW_API=1 \`
`       gcr.io/vorvan/h2oai/h2ogpt-runtime:latest /workspace/generate.py \`
`          --score_model=None \`
`          --max_output_seq_len=8192 \`
`          --max_seq_len=8192 \`
`          --prompt_type=llama3 \`
`          --use_safetensors=True \`
`          --base_model=meta-llama/Meta-Llama-3.1-8B \`
`          --tokenizer-path=meta-llama/Meta-Llama-3.1-8B \`
`          --save_dir='/workspace/save/' \`
`          --use_auth_token=$HUGGING_FACE_HUB_TOKEN \`
`          --use_gpu_id=False \`
`          --allow_upload_to_user_data=False \`
`          --allow_upload_to_my_data=True \`
`          --enable_ocr='off' \`
`          --enable_pdf_ocr='off' \`
`          --langchain_mode='UserData' \`
`          --langchain_modes="['LLM', 'UserData', 'MyData']" \`
`          --user_path=/workspace/user_path \`
`          --db_type=chroma \`
`          --visible_h2ogpt_header=False \`
`          --visible_doc_selection_tab=False \`
`          --visible_doc_view_tab=False \`
`          --visible_chat_history_tab=False \`
`          --visible_expert_tab=False \`
`          --visible_models_tab=False \`
`          --visible_system_tab=False \`
`          --visible_tos_tab=False \`
`          --visible_hosts_tab=False \`
`          --answer_with_sources=False \`
`        --visible_side_bar=True \`
`          --visible_login_tab=False \`
`          --visible_visible_models=False \`
`        --visible_submit_buttons=True \`
`        --visible_chat_tab=True \`
`          --visible_h2ogpt_links=False \`
`          --visible_h2ogpt_qrcode=False \`
`          --visible_h2ogpt_logo=False \`
`          --visible_chatbot_label=False`
` # EOF`
`root@ai-ubuntu22gpu-gpt:/opt#`

beste regards
Daniel Plominski

**Comment by pseudotensor** [comment]
Try not passing prompt_type. It is not required.

**Comment by plitc** [comment]
without prompt_type =  the model only spits out confusing stuff

Logoutput

`INFO:     172.17.0.1:40054 - "GET /queue/data?session_hash=e0apa8hwqu HTTP/1.1" 200 OK
No chat template is set for this tokenizer, falling back to a default class-level template. This is very error-prone, because models are often trained with templates different from the class default! Default chat templates are a legacy feature and will be removed in Transformers v4.43, at which point any code depending on them will stop working. We recommend setting a valid chat template before then to ensure that this model continues working without issues.
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:128001 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token.As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
INFO:     172.17.0.1:43008 - "GET / HTTP/1.1" 200 OK
`

**Comment by plitc** [comment]
[pseudotensor](https://github.com/pseudotensor)

can you please list a Docker example that exactly reflects the functionality of https://gpt.h2o.ai/?

**Comment by pseudotensor** [comment]
You are using the base model, you need to use the instruct version of the model.

**Comment by plitc** [comment]
Does not work:

`          --prompt_type=instruct \`
`          --use_safetensors=True \`
`          --base_model=meta-llama/Meta-Llama-3.1-8B-instruct \`
`          --tokenizer-path=meta-llama/Meta-Llama-3.1-8B-instruct \`

Confusing output:

![h2ogpt](https://github.com/user-attachments/assets/4d3f6ddc-8f3e-4df7-b221-e47916f72bb6)

**Comment by pseudotensor** [comment]
You are using --prompt_type=instruct.  Please do not pass prompt_type, like we discussed.

**Comment by pseudotensor** [comment]
Also there is no such parameter as `tokenizer-path` perhaps you meant tokenizer_base_model, but that's not required.

**Comment by plitc** [comment]
Okey, without:

`--prompt_type=instruct \`
`--tokenizer-path=meta-llama/Meta-Llama-3.1-8B-instruct \`

it now seems to deliver better results

**Comment by pseudotensor** [comment]
I'll share later how I launch gpt.h2o.ai.  The previous version (which is similar) is here already: https://github.com/h2oai/h2ogpt/blob/main/docs/FAQ.md#deploying-like-gpth2oai
